c7n/resources/ec2.py

```diff
@@ -388,37 +388,34 @@ def process_instance(i):
         with self.executor_factory(max_workers=2) as w:
             list(w.map(process_instance, instances))
 
-@actions.register('create-snapshot')
+@actions.register('snapshot')
 class Snapshots(BaseAction):
     def process(self, resources):
-        volume_sets = {}
         for resource in resources:
-            volume_sets[resource['InstanceId']] = []
-            for blockDevice in resource['BlockDeviceMappings']:
-                volume_sets[resource['InstanceId']].append(blockDevice['Ebs']['VolumeId'])
-        with self.executor_factory(max_workers=3) as w:
-            futures = []
-            for volume_set in volume_sets:
-                futures.append(w.submit(self.process_volume_set, volume_set, volume_sets[volume_set]))
+            with self.executor_factory(max_workers=3) as w:
+                futures = []
+                futures.append(w.submit(self.process_volume_set, resource))
                 for f in as_completed(futures):
                     if f.exception():
                         self.log.error("Exception creating snapshot set \n %s" % (f.exception()))
+            
+            
 
-    def process_volume_set(self, instanceId, volume_set):
+    def process_volume_set(self, resource):
         c = utils.local_session(self.manager.session_factory).client('ec2')
-        r = utils.local_session(self.manager.session_factory).resource('ec2')
-        for volumeId in volume_set:
-            snapDescription = "Automated,LocalBackup,%s,%s" % (instanceId, volumeId)
-            response = c.create_snapshot(DryRun=True, VolumeId=volumeId, Description=snapDescription)
-            snapShot = r.Snapshot(response['SnapshotId'])
-            snapshot.create_tags(
-                DryRun=True, 
-                Tags=[
-                    {'Key' : 'Name','Value' : volumeId},
-                    {'Key' : 'InstanceId','Value' : instanceId},
-                ]
-            )
-            
+        for blockDevice in resource['BlockDeviceMappings']:
+            if 'Ebs' in blockDevice:
+                snapDescription = "Automated,LocalBackup,%s,%s" % (resource['InstanceId'], block_device['Ebs']['VolumeId'])
+                response = c.create_snapshot(DryRun=False, VolumeId=block_device['Ebs']['VolumeId'], Description=snapDescription)
+                c.create_tags(
+                    DryRun=False, 
+                    Resources=[response['SnapshotId']],
+                    Tags=[
+                        {'Key' : 'Name','Value' : volumeId},
```

Comment:
There's a few minor style things here, not sure which editor your using, but using flake8 is very helpful to adhering to python code style guide as well picking up and integrates with most editors/ides.